lifecycle-coroutines/src/commonMain/kotlin/com/arkivanov/essenty/lifecycle/coroutines/FlowWithLifecycle.kt

```diff
@@ -0,0 +1,33 @@
+package com.arkivanov.essenty.lifecycle.coroutines
+
+import com.arkivanov.essenty.lifecycle.Lifecycle
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+
+/**
+ * Start [Flow] collecting when [Lifecycle.State] is at least as [minActiveState].
+ * It stopped collecting if "opposite" [Lifecycle.State] will appear.
+ */
+fun <T> Flow<T>.withLifecycle(
+    lifecycle: Lifecycle,
+    minActiveState: Lifecycle.State = Lifecycle.State.STARTED
+): Flow<T> = callbackFlow {
```

Comment:
It looks like we also need to pass `Dispatcher` here as well, as a 3rd parameter, with default value also be `Dispatchers.Main`. Otherwise it won't be possible to use `Dispatchers.Main.immediate` on supporting platforms.

Reply:
Added.

Reply:
Thank you! Looks like I missed the fact that the argument could be `CoroutineContext` instead of `CoroutineDispatcher`. Would it be possible for you to change both arguments to `context: CoroutineContext = Dispatchers.Main`?